src/index.js

```diff
@@ -33,13 +43,17 @@ class ReactImageFallback extends React.Component {
 ReactImageFallback.displayName = "ReactImageFallback";
 
 ReactImageFallback.propTypes = {
-	src: React.PropTypes.string.isRequired,
-	fallbackImage: React.PropTypes.string.isRequired,
-	initialImage: React.PropTypes.string,
+	src: PropTypes.string.isRequired,
+	fallbackImage: PropTypes.string.isRequired,
+	initialImage: PropTypes.string,
+	shouldUpdateImage: PropTypes.func
 };
 
 ReactImageFallback.defaultProps = {
-	initialImage: null
+	initialImage: null,
+	shouldUpdateImage: function(props, nextProps) {
+		return props.src !== nextProps.src;
```

Comment:
Ahh. this is an interesting idea. it gives you some really unique power as a parent. ill think about this one.